data/pigui/views/game.lua

```diff
@@ -118,35 +90,30 @@ local function displayOnScreenObjects()
 		local mp = ui.getMousePos()
 		-- mouse release handler for radial menu
 		if (mp - mainCoords):length() < click_radius then
-			if not ui.isAnyWindowHovered() and ui.isMouseClicked(1) then
+			if ui.canClickOnScreenObjectHere() and ui.isMouseClicked(1) then
 				local body = mainBody
 				ui.openDefaultRadialMenu(body)
 			end
 		end
 		-- mouse release handler
 		if (mp - mainCoords):length() < click_radius then
-			if not ui.isAnyWindowHovered() and ui.isMouseReleased(0) then
+			if ui.canClickOnScreenObjectHere() and ui.isMouseReleased(0) then
 				if group.hasNavTarget or combatTarget == mainBody then
 					-- if clicked and is target, unset target
 					if group.hasNavTarget then
 						player:SetNavTarget(nil)
 					else
 						player:SetCombatTarget(nil)
 					end
-					-- if not in setspeed mode or ctrl-click and is setspeed target,
-					-- clear setspeed target
-					if not player:GetSetSpeed() or (ui.ctrlHeld() and group.hasSetSpeedTarget) then
-						player:SetSetSpeedTarget(nil)
-					end
 				elseif not group.multiple then
 					-- clicked on single, just set navtarget/combatTarget
 					setTarget(mainBody)
 					if ui.ctrlHeld() then
-						-- also set setspeed target on ctrl-click
-						player:SetSetSpeedTarget(mainBody)
-					elseif not player:GetSetSpeed() then
-						-- clear setspeed target if not in setspeed mode
-						player:SetSetSpeedTarget(nil)
+						-- also set follow target on ctrl-click
+						player:SetFollowTarget(mainBody)
+					elseif not player:GetCruiseSpeed() then
+						-- clear follow target if not in cruise mode XXX???
+						-- player:SetFollowTarget(nil)
```

Comment:
I'd like to see this branch removed or the comment resolved before merge for cleanliness.

Reply:
Done!